### ISSUE TITLE
wifi not working on mbp 14,1

### ISSUE THREAD

**Issue by pranavg189** [Dunedan/mbp-2016-linux#56]
Hello,

I installed Manjaro on my mbp 14,1 and the wifi worked initially for some time but now the 'brcmfmac' module is failing to load for me with the following dmesg error:

`brcmfmac: brcmf_chip_recognition: chip backplane type 15 is not supported`
`brcmfmac: brcmf_pcie_probe: failed 14e4:43a3`

I tried installing all possible drivers and solutions for the Broadcom BCM4350 wifi chipset given here:

https://wiki.archlinux.org/index.php/broadcom_wireless
https://bbs.archlinux.org/viewtopic.php?id=216743
https://bbs.archlinux.org/viewtopic.php?id=218310
https://github.com/Dunedan/mbp-2016-linux/issues/10

Surprisingly, when I boot into the manjaro live usb, wifi works fine. But If I reload the 'brcmfmac' module in the live usb, it again gives the above error and wifi stops working.

I have tried the following versions of linux kernel: 4.14, 4.16

Everything else is working great.

Has anyone got the wifi working on mbp 14,1 ? Seems related to #10 

It would be great if some note about wifi on mbp 14,1 could be added in the 'Wifi' section of README.md

**Comment by craigerrington** [comment]
Apologies, misread your model.

**Comment by pranavg189** [comment]
> I'm on a MacBookPro14,1 running Ubuntu 17.10 and wifi worked out of the box, and I haven't experienced any issues at all (there are no dropouts, I can connect to 5GHz networks, etc).

Maybe it is an issue with 'brcmfmac' firmware shipped with Manjaro. Can anyone confirm the working of mbp 14,1 wifi on other distributions like Archlinux, Fedora etc ?

**Comment by pranavg189** [comment]
Update: I just installed Ubuntu 18.04 on my mbp 14,1 and the wifi is working properly now along with the applespi (touchpad and keyboard) driver.

Another thing I noticed is that, If I login into "Ubuntu on Wayland" session, the wifi stops working as it used to in Manjaro and can not be restarted again. I had to reboot and select the normal "Ubuntu" session. Hence, I think logging into the Wayland session currently has some issue with the "brcmfmac" driver which disables the wifi.

Hope this helps someone installing linux on mbp 14,1 .

It would be great if some note is added in the README.md file about this. I will confirm again if this really is the problem. I will need confirmation from other users too if Wayland has issues with brcmfmac driver.
